mobile/lib/shared/services/sync.service.dart

```diff
@@ -62,6 +62,15 @@ class SyncService {
   }) =>
       _lock.run(() => _syncRemoteAlbumsToDb(remote, isShared, loadDetails));
 
+  /// Syncs single remote album to the database
+  /// returns `true` if there were any changes
+  Future<bool> syncRemoteAlbumToDb(
+    AlbumResponseDto remote,
+    Album album, {
+    required FutureOr<AlbumResponseDto> Function(AlbumResponseDto) loadDetails,
+  }) =>
+      _lock.run(() => _syncRemoteAlbum(remote, album, [], [], loadDetails));
```

Comment:
This won't handle deletes at all and result in a garbage DB state.

Reply:
Have you tested it? It works fine for me. Where supposed to be a problem?

Reply:
Well, this function can also be called for shared albums. So it needs to handle the delete candidates.

Reply:
> Well, this function can also be called for shared albums. So it needs to handle the delete candidates.

Correct, therefore I don't know what problem do you see in my implementation. I will fix it if you can help me find what's wrong.

Reply:
Let's have a shared album A owned by user O.
User O has shared A with user U.
User U opens the app, it syncs the asset and albums from the server. O removes assets from A, U adds a new asset to A. `syncRemoteAlbum` is called for this individual album A. The deletes are not correctly handled and remain in the database of user U's app although they maybe should not.

Reply:
If you're plannig bigger changes that comprise solving this issue, feel free to close it in favour of your rework. Anyway thanks for giving me the bigger picture.

Reply:
we could run a cleanup job once per day (or delayed on app start) to find assets not accessible by the logged-in user.

however, I just found another issue with the single album refresh :/ unless we find a solution, I'd propose to change this PR back into a draft.
In any case: Thank you for your work! Also for identifying that new bug...

Reply:
> Let's have a shared album A owned by user O. User O has shared A with user U. User U opens the app, it syncs the asset and albums from the server. O removes assets from A, U adds a new asset to A. `syncRemoteAlbum` is called for this individual album A. The deletes are not correctly handled and remain in the database of user U's app although they maybe should not.

It's not only update among multiple users, even updates from different clients of the same users do that. When an asset is added into album on web mobile doesn't sync album dates. Maybe album versioning would help. Each album change would raise album version number and clients can easily recognize changes and sync album no matter asset was added or deleted.